src/country_finder.rs

```diff
@@ -25,27 +25,28 @@ impl Default for CountryFinder {
     }
 }
 
-impl<'a> FromIterator<&'a Zone> for CountryFinder {
-    fn from_iter<I: IntoIterator<Item = &'a Zone>>(zones: I) -> Self {
+impl CountryFinder {
+    pub fn init(zones: &[Zone], typer: &ZoneTyper) -> Self {
         CountryFinder {
             countries: zones
                 .into_iter()
                 .filter_map(|z| {
-                    z.tags.get(COUNTRY_CODE_TAG).map(|country_code| {
-                        (
-                            z.id.clone(),
-                            Country {
-                                iso: country_code.clone(),
-                                admin_level: z.admin_level.clone(),
-                            },
-                        )
-                    })
+                    z.tags
+                        .get(COUNTRY_CODE_TAG)
+                        .filter(|ccode| typer.countries_rules.contains_key(*ccode))
```

Comment:
hum it would maybe be more encapsulated to add a `contains_rule` function to the typer ?